internal/provider/certificate_resource.go

```diff
@@ -373,6 +383,18 @@ func (r *CertificateResource) Create(ctx context.Context, req resource.CreateReq
 		return
 	}
 
+	// Check that certificates are successfully added to Third Parties
+	thirdParties := make([]string, 0, len(data.ThirdParties.Elements()))
```

Comment:
I think we should explore using the [Terraform's built-in retry mechanism](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts) for provisioning certificates with third parties. It seems made for this use case and better integrated overall.